src/modules/robot/arm_solutions/RotatableDeltaSolution.cpp

```diff
@@ -0,0 +1,253 @@
+     float z1 = -delta_rf*sinf(theta1);
+
+     float y2 = (t + delta_rf*cosf(theta2))*sin30;
+     float x2 = y2*tan60;
+     float z2 = -delta_rf*sinf(theta2);
+
+     float y3 = (t + delta_rf*cosf(theta3))*sin30;
+     float x3 = -y3*tan60;
+     float z3 = -delta_rf*sinf(theta3);
+
+     float dnm = (y2-y1)*x3-(y3-y1)*x2;
+
+     float w1 = y1*y1 + z1*z1;
+     float w2 = x2*x2 + y2*y2 + z2*z2;
+     float w3 = x3*x3 + y3*y3 + z3*z3;
+
+     // x = (a1*z + b1)/dnm
+     float a1 = (z2-z1)*(y3-y1)-(z3-z1)*(y2-y1);
+     float b1 = -((w2-w1)*(y3-y1)-(w3-w1)*(y2-y1))/2.0F;
+
+     // y = (a2*z + b2)/dnm;
+     float a2 = -(z2-z1)*x3+(z3-z1)*x2;
+     float b2 = ((w2-w1)*x3 - (w3-w1)*x2)/2.0F;
+
+     // a*z^2 + b*z + c = 0
+     float a = a1*a1 + a2*a2 + dnm*dnm;
+     float b = 2.0F*(a1*b1 + a2*(b2-y1*dnm) - z1*dnm*dnm);
+     float c = (b2-y1*dnm)*(b2-y1*dnm) + b1*b1 + dnm*dnm*(z1*z1 - delta_re*delta_re);
+
+     // discriminant
+     float d = b*b - (float)4.0F*a*c;
+     if (d < 0.0F) return -1; // non-existing point
+
+     z0 = -(float)0.5F*(b+sqrtf(d))/a;
+     x0 = (a1*z0 + b1)/dnm;
+     y0 = (a2*z0 + b2)/dnm;
+
+     z0 += z_calc_offset; //nj
+     return 0;
+}
+
+
+void RotatableDeltaSolution::init() {
+
+	z_calc_offset  = (delta_z_offset - tool_offset - delta_ee_offs)*-1.0F;
+
+    // This is calculated from the Z_HOME_ANGLE angles specified in the config (file), after applying forward
+    // kinematics, and adding the Z calc offset to it.
+	z_home_offs    = (((delta_z_offset - tool_offset - delta_ee_offs) - 182.002F) - 0.5F);
+
+}
+
+void RotatableDeltaSolution::cartesian_to_actuator( float cartesian_mm[], float actuator_mm[] )
+{
+	//We need to translate the Cartesian coordinates in mm to the actuator position required in mm so the stepper motor  functions
+	float alpha_theta = 0.0F;
+	float beta_theta  = 0.0F;
+	float gamma_theta = 0.0F;
+
+	//Code from Trossen Robotics tutorial, note we put the X axis at the back and not the front of the robot.
+
+	float x0 = cartesian_mm[X_AXIS];
+	float y0 = cartesian_mm[Y_AXIS];
+	float z_with_offset = cartesian_mm[Z_AXIS] + z_calc_offset; //The delta calculation below places zero at the top.  Subtract the Z offset to make zero at the bottom.
+
+	int status =              delta_calcAngleYZ(x0,                    y0,                  z_with_offset, alpha_theta);
+	if (status == 0) status = delta_calcAngleYZ(x0*cos120 + y0*sin120, y0*cos120-x0*sin120, z_with_offset, beta_theta);  // rotate co-ordinates to +120 deg
+	if (status == 0) status = delta_calcAngleYZ(x0*cos120 - y0*sin120, y0*cos120+x0*sin120, z_with_offset, gamma_theta);  // rotate co-ordinates to -120 deg
+
+	if (status == -1)  //something went wrong
+	{
+	    //Do nothing so the actuator_mm is unchanged by this routine.
+            // as we have no idea what it should be
+//DEBUG CODE, uncomment the following to help determine what may be happening if you are trying to adapt this to your own different roational delta.
```

Comment:
actuator_mm is not set to anything on call so needs to be written to otherwise it will have random data in it.